playbooks/roles/tarsnap/tasks/tarsnap.yml

```diff
@@ -0,0 +1,70 @@
+---
+# Parts of this file are based on:
+#   url: https://github.com/al3x/sovereign/blob/a48a9941009cc1f18450d01cc7cf4f067ece3fae/roles/tarsnap/tasks/tarsnap.yml
+#   contributors:  al3x jplock lvillani lukecyca jsravn bcachet alexdunae pdebruic
+# The sovereign project source and this derivative work are licensed under the GNU General Public License v3.
+
+- name: Install dependencies for Tarsnap
+  apt: pkg={{ item }} state=installed
+  with_items:
+    - e2fslibs-dev
+    - libssl-dev
+    - zlib1g-dev
+    - build-essential
+    - python-setuptools
+
+- name: Check if tarsnap {{ tarsnap_version }} is installed
+  shell: tarsnap --version | grep {{ tarsnap_version }} --color=never
+  register: tarnsap_installed
+  changed_when: "tarnsap_installed.stderr != ''"
+  ignore_errors: yes
+
+- name: Download the current tarsnap code signing key
+  when: tarnsap_installed|failed
+  get_url:
+    url=https://www.tarsnap.com/tarsnap-signing-key.asc
+    dest=/root/tarsnap-signing-key.asc
+
+- name: Add the tarsnap code signing key to your list of keys
+  when: tarnsap_installed|failed
+  command:
+    gpg --import tarsnap-signing-key.asc
+    chdir=/root/
+
+- name: Download tarsnap SHA file
+  when: tarnsap_installed|failed
+  get_url:
+    url="https://www.tarsnap.com/download/tarsnap-sigs-{{ tarsnap_version }}.asc"
+    dest="/root/tarsnap-sigs-{{ tarsnap_version }}.asc"
+
+- name: get the SHA256sum for this tarsnap release
+  when: tarnsap_installed|failed
+  shell: >
+    gpgResult=`gpg --decrypt tarsnap-sigs-{{ tarsnap_version }}.asc 2>/dev/null`;
+    echo ${gpgResult#*=};
+  args:
+    executable: /bin/bash
+    chdir: /root
+  register: tarsnap_sha
+
+- name: Download Tarsnap source
+  when: tarnsap_installed|failed
+  get_url:
+    url="https://www.tarsnap.com/download/tarsnap-autoconf-{{ tarsnap_version }}.tgz"
+    dest="/root/tarsnap-autoconf-{{ tarsnap_version }}.tgz"
+    sha256sum={{ tarsnap_sha.stdout_lines[0] }}
+
+- name: Decompress Tarsnap source
+  when: tarnsap_installed|failed
+  command: tar xzf /root/tarsnap-autoconf-{{ tarsnap_version }}.tgz chdir=/root creates=/root/tarsnap-autoconf-{{ tarsnap_version }}/COPYING
+
+- name: Configure Tarsnap for local build
+  when: tarnsap_installed|failed
+  command: ./configure chdir=/root/tarsnap-autoconf-{{ tarsnap_version }} creates=/root/tarsnap-autoconf-{{ tarsnap_version }}/Makefile
+
+- name: Build and install Tarsnap
+  when: tarnsap_installed|failed
+  command: make all install clean chdir=/root/tarsnap-autoconf-{{ tarsnap_version }}
```

Comment:
You seem to have removed the `creates` setting here, so this will now always be run, even when the tarsnap binary already exists.  Is there any specific reason why you did this?

Reply:
Right @UmanShahzad, there's already a guard against running this excessively. If the correct version is already installed, `tarsnap_installed` will be set, and this won't run.

With the `creates` setting there was no way to update tarsnap, except I guess removing the binary before running the playbook, but that didn't seem like a good solution.